larq_compute_engine/tests/end2end_test.py

```diff
@@ -152,43 +131,67 @@ def toy_model_int8(**kwargs):
 
 
 def preprocess(data):
-    return lqz.preprocess_input(data["image"])
+    return lqz.preprocess_input(data["image"]), data["label"]
 
 
-def assert_model_output(model_lce, inputs, outputs):
+def assert_model_output(model_lce, inputs, outputs, rtol, atol):
     interpreter = Interpreter(model_lce, num_threads=min(os.cpu_count(), 4))
     actual_outputs = interpreter.predict(inputs)
-    np.testing.assert_allclose(actual_outputs, outputs, rtol=0.001, atol=0.25)
+    np.testing.assert_allclose(actual_outputs, outputs, rtol=rtol, atol=atol)
 
 
 @pytest.mark.parametrize(
     "model_cls",
     [toy_model, toy_model_sequential, toy_model_int8, lqz.sota.QuickNet],
 )
 def test_simple_model(model_cls):
-    model = model_cls(weights="imagenet")
-    model_lce = convert_keras_model(
-        model, experimental_enable_bitpacked_activations=True
-    )
-
-    # Test on the flowers dataset
+    # Test on the OxfordFlowers dataset
     dataset = (
         tfds.load("tf_flowers", split="train", try_gcs=True)
         .map(preprocess)
         .shuffle(100)
         .batch(10)
-        .take(1)
     )
-    inputs = next(tfds.as_numpy(dataset))
 
+    model = model_cls(weights="imagenet")
+
+    # For the untrained models, do a very small amount of training so that the
+    # batch norm stats (and, less importantly, the weights) have sensible
+    # values.
+    if model_cls != lqz.sota.QuickNet:
+        model.compile(
+            optimizer=tf.keras.optimizers.Adam(1e-4),
+            loss="sparse_categorical_crossentropy",
+        )
+        model.fit(dataset, epochs=1, steps_per_epoch=10)
+
+    model_lce = convert_keras_model(
+        model, experimental_enable_bitpacked_activations=True
+    )
+
+    if model_cls == lqz.sota.QuickNet:
+        # Since QuickNet is a deep model we are more tolerant of error.
+        rtol = 0.05
+        atol = 0.2
+    elif model_cls == toy_model_int8:
+        # 0.025 atol is chosen because it allows an off-by-one error in the int8
+        # model (with +-3 scales) but not off-by-two.
+        rtol = 0.01
+        atol = 0.025
+    else:
+        rtol = 0.001
+        atol = 0.001
```

Comment:
@Tombana yeah you're right, turns out we can use *way* lower tolerances for the non-QuickNet models. And probably should do.